hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/BlockRecordManagerImpl.java

```diff
@@ -281,6 +281,27 @@ public Bytes blockHashByBlockNumber(final long blockNo) {
         return BlockRecordInfoUtils.blockHashByBlockNumber(lastBlockInfo, blockNo);
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public void advanceConsensusClock(@NonNull final Instant consensusTime, @NonNull final HederaState state) {
+        final var builder = this.lastBlockInfo
+                .copyBuilder()
+                .consTimeOfLastHandledTxn(Timestamp.newBuilder()
+                        .seconds(consensusTime.getEpochSecond())
+                        .nanos(consensusTime.getNano()));
+        final var newBlockInfo = builder.build();
+
+        // Update the latest block info in state
+        final var states = state.createWritableStates(BlockRecordService.NAME);
+        final var blockInfoState = states.<BlockInfo>getSingleton(BlockRecordService.BLOCK_INFO_STATE_KEY);
+        blockInfoState.put(newBlockInfo);
+        // Commit the changes. We don't ever want to roll back when advancing the consensus clock
+        ((WritableSingletonStateBase<BlockInfo>) blockInfoState).commit();
```

Comment:
@Neeharika-Sompalli @tinker-michaelj I want to check, this comment is correct?

Reply:
Makes sene to me Matt.

Reply:
thanks!